packages/core/src/lint.ts

```diff
@@ -89,3 +91,46 @@ export async function lintDocument(opts: {
   });
   return ctx.problems.map((problem) => config.addProblemToIgnore(problem));
 }
+
+export async function LintDocumentConfig(opts: {
+  document: Document
+}) {
+  const { document } = opts;
+  const externalRefResolver = new BaseResolver();
+  const oasVersion = OasVersion.Version3_0; // set default version for config files.
+
+  const config = new LintConfig({
+    plugins: [defaultPlugin],
+    extends: [],
+    rules: { spec: 'error' },
+  });
+
+  const ctx: WalkContext = {
+    problems: [],
+    oasVersion: oasVersion,
+  };
+
+  // TODO: maybe need deprecate this part with preprocessors and regularRules.
+  const oasMajorVersion = openAPIMajor(oasVersion);
+  const rules = config.getRulesForOasVersion(oasMajorVersion);
+  const types = normalizeTypes(configTypes, config);
+  const preprocessors = initRules(rules as any, config, 'preprocessors', oasVersion);
+  const regularRules = initRules(rules as any, config, 'rules', oasVersion);
+  const normalizedVisitors = normalizeVisitors([...preprocessors, ...regularRules], types);
```

Comment:
This is too verbose for simple case and also would be great to skip oasVersion hack.

Try this:

```suggestion
  // import specRule somewhere
  const rules = [ { severity: 'error', ruleId: 'spec', visitor: specRule({severity: 'error'}) } ];
  const normalizedVisitors = normalizeVisitors(rules, types);
```

Looks like you won't need config at all too.

Reply:
Thanks, it very helped me. I thought about it but couldn't implement it.